### PR TITLE
Remove setup.py

### PR DESCRIPTION
**Pull request by dstansby**
Two misc cleanups to `setup.py`:

- Specify the extras using `setup.cfg`, instead of using python code
- Use `pyproject.toml` for `setuptools_scm`.

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -1,33 +1,6 @@
...
-ex_extras = dict(filter(lambda i: i[0] not in exclude_keys, extras.items()))
-# Concatenate all the values together for 'all'
-extras['all'] = list(chain.from_iterable(ex_extras.values()))
-
-setup(
-    extras_require=extras,
-    use_scm_version={'write_to': os.path.join('sunpy', '_version.py')},
-    ext_modules=get_extensions(),
-)
+setup(ext_modules=get_extensions())
```
Comment (Cadair) [inline setup.py:6]: We don't even need this, you can do:
```
[extension-helpers]
use_extension_helpers = true
```

https://extension-helpers.readthedocs.io/en/latest/using.html

**Comment by Cadair** [comment]
I think it's probably best if we keep a barebones setup.py file around for a little while longer. There's a discussion over on astropy with some reasons. I don't really see any downsides to doing so if we have no code or config in there.

**Comment by dstansby** [comment]
👍 , I'm guessing https://github.com/astropy/astropy/pull/14361 is the xref for anyone interested

**Comment by nabobalis** [comment]
So we should leave this for 5.1?

**Comment by dstansby** [comment]
I think we can put it in, I just need to put a `setup.py` file back that calls `setup()`.

**Review by Cadair — APPROVED** [review]
lgtm, the next thing to do is move all the setup.cfg stuff to pyproject.toml

**Pull request merged**